### PR TITLE
advanced-search messages

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.html [hunk 1]
```diff
@@ -103,3 +103,5 @@
     <i class="icon-search-button" [attr.tooltip-value]="'search' | translate" tooltip-class="bottom"></i>
 </div>
 
+<div class="message" [class.hide]="!showMessage">Please select search parameters</div>
```
Comment (tltk90) [inline src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.html:106]: add `translate` pipe.

File: src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.ts [hunk 2]
```diff
@@ -97,8 +99,20 @@ import { SearchAction } from '../../actions/status-bar.actions';
...
+		if (this.isValid(params)) {
+			this.store.dispatch(new SearchAction(params));
+			this._parent.close();
+		} else {
+			this.showMessage = true;
+		}
+	}
+
+	isValid(params: IAdvancedSearchParameter): boolean {
+		return params?.registeration?.length > 0
```
Comment (tltk90) [inline src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.ts:112]: I thinks it's ok that the registration be empty.

File: src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.spec.ts [hunk 3]
```diff
@@ -87,4 +97,44 @@ describe('AdvancedSearchComponent', () => {
...
+		it('should trigger a search if parameters are valid', () => {
+			spyOn(component, 'isValid').and.returnValue(true);
+			spyOn(store, 'dispatch');
+			spyOn(parent, 'close');
+			component.search();
+			expect(store.dispatch).toHaveBeenCalled();
+			expect(parent.close).toHaveBeenCalled();
+		});
+
+	})
```
Comment (tltk90) [inline src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.spec.ts:139]: 💯

**Review by tltk90 — CHANGES_REQUESTED** [review]

File: src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.ts (hunk 2 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.ts:112]: This is not what I'm checking here.  Validation will fail only if **all** search fields are empty.

File: src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.html (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.html]: done

**Review by tltk90 — APPROVED** [review]

File: src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.spec.ts (hunk 3 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/status-bar/components/advanced-search/advanced-search.component.spec.ts:139]: 👍

**Pull request merged**